app/routines/update_user_salesforce_info.rb

```diff
@@ -0,0 +1,148 @@
+class UpdateUserSalesforceInfo
+  USER_BLOCK_SIZE = 1000
+
+  def initialize
+    @errors = []
+  end
+
+  def self.call
+    new.call
```

Comment:
No not giving up, just in these cases where we don't want a big transaction, and we aren't nesting this in another routine or calling other routines from it, and we aren't using lev's error functionality, I don't see as much of a need.